scripts/ci/jobs/diff-dumps.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../../.. && pwd)"
+source "$ROOT/scripts/ci/gcp.sh"
+source "$ROOT/scripts/ci/lib.sh"
+
+set -euo pipefail
+
+create_diff_dumps() {
+    info "Creating diff dump for each manifest"
+
+    require_executable "jq"
+
+    local idx
+
+    mkdir -p /tmp/diff-dumps
+    idx=-1
+    while IFS=$'\t' read -r dumploc timestamp config; do
+        idx=$((idx+1))
+        dump_file_name="${dumploc##*/}"
+        echo "Pulling genesis dump from ${dumploc}"
+        gsutil cp "${dumploc}" .
+        timestamp_in_zip="$(unzip -p "${dump_file_name}" manifest.json | jq -r '.until')"
+        echo "Got timestamps -- from zip: ${timestamp_in_zip}; from manifest: ${timestamp}"
+        [[ "${timestamp_in_zip}" == "${timestamp}" ]] # Assertion on the manifest contents
+        # ./bin/updater is from the generate-genesis image in OpenShift CI.
+        ./bin/updater diff-dumps --base-dump "${dump_file_name}" --head-dump /tmp/genesis-dump/dump.zip --config "${config}" --out-file "/tmp/diff-dumps/dump${idx}/diff.zip"
+    done < <(jq -r '.knownGenesisDumps | .[]| [.dumpLocationInGS, .timestamp, (.config // empty | tostring)] | @tsv' < image/scanner/dump/genesis_manifests.json)
+    du -d 2 -kh "/tmp/diff-dumps"
+}
+
+upload_diff_dumps() {
+    info "Uploading diff dumps"
+
+    local branch
+    local idx
+
+    branch=$(get_base_ref)
+    idx=-1
+    while IFS=$'\t' read -r diffUUID; do
+        idx=$((idx+1))
+        expected_zip_file_loc="/tmp/diff-dumps/dump${idx}/diff.zip"
+        [[ -f "${expected_zip_file_loc}" ]]
+        if [[ -z "${diffUUID}" ]]; then
+            continue
+        fi
+        echo "Found file at ${expected_zip_file_loc}"
+        du -skh "${expected_zip_file_loc}"
+        cmd=()
+        # Note: CIRCLE_TAG is defined via openshift_ci_mods in scripts/ci/lib.sh.
+        if [[ "$branch" != "master" && -z "${CIRCLE_TAG}" ]]; then
```

Comment:
@gavin-stackrox I pretty blindly took this from circleci expect I set branch to `branch=$(get_base_ref)` instead of `CRICLE_BRANCH`. Do you think this is an ok way to check if this is master or tag or do we have a better way that I missed?